Assets/MRTK/Core/Interfaces/EventSystem/IMixedRealityEventPropagationData.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace Microsoft.MixedReality.Toolkit
+{
+    /// <summary>
+    /// Direction of event propagation through a hierarchy of elements
+    /// </summary>
+    [System.Flags]
+    public enum EventPropagation // Mainly used to build Paths
+    {
+        None = 0,
+        BubblesUp = 1, // Target up to root
+        TricklesDown = 2, // Root down to target
+        Cancellable = 4, // ?
```

Comment:
Slight nit, but we've tended to recommend using bit shifts for defining flags. Makes it a bit more obvious that you still have all values accounted for when the shifts all increment by 1.

https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/CodingGuidelines.html#do-8

```suggestion
        None = 0 << 0,
        BubblesUp = 1 << 0, // Target up to root
        TricklesDown = 1 << 1, // Root down to target
        Cancellable = 1 << 2, // ?
```

Reply:
Thanks for pointing this out @keveleigh .